### PR TITLE
Remove `propdict`

### PR DESCRIPTION
**Pull request by fhagemann**
The method `propdict` is basically an alias of `PropDicts.readprops`.
Thinking about an upcoming breaking release of `LegendGeSim`, `propdict` should be removed in favor of `readprops`.
I have deprecated `propdict` in this PR.

### CODE REVIEW

**Comment by fhagemann** [comment]
Also looks like `propdict` was not exported, so it should affect too many people hopefully

**Comment by fhagemann** [comment]
The last commit is cherry-picked from #63 (be83000), should resolve the failing tests.

File: src/detector.jl [hunk 1]
```diff
@@ -190,8 +190,8 @@ function construct_ssd_simulation(det_meta::PropDict, env::Environment, simulato
         error("Only 2D is supported up to now.")
     end
     sim.world = begin
-        crystal_radius = to_SSD_units(T, det_meta.geometry.radius_in_mm, u"mm")
-        crystal_height = to_SSD_units(T, det_meta.geometry.height_in_mm, u"mm")
+        crystal_radius = SolidStateDetectors._parse_value(T, det_meta.geometry.radius_in_mm, u"mm")
```
Comment (oschulz) [inline src/detector.jl]: We shouldn't use internal functions of SolidStateDetectors.

If we need that, we should make it part of the public SolidStateDetectors API.

File: src/temp_utils.jl [hunk 2]
```diff
@@ -11,21 +11,7 @@ E.g. filename("/some/path/to/file_name.ext") -> "file_name"
...
-Construct a PropDict based on given <json_file>.
-Alias for `readprops` from PropDicts.jl
-    
-"""
-function propdict(json_file::AbstractString)
-    readprops(json_file)
-end
-
-    
+Base.@deprecate propdict(json_file::AbstractString) readprops(json_file)
```
Comment (oschulz) [inline src/temp_utils.jl]: If users of LegendGeSim don't typically use this, let's remove it completely.

**Review by oschulz — CHANGES_REQUESTED** [review]

File: src/detector.jl (hunk 1 above)
Comment (fhagemann) [reply src/detector.jl]: Ok, I'm removing this from this PR then.

File: src/temp_utils.jl (hunk 2 above)
Comment (fhagemann) [reply src/temp_utils.jl]: I force-pushed to remove `propdict` completely in this PR.

**Comment by fhagemann** [comment]
@oschulz Please check and merge at your earliest convenience.

File: src/detector.jl [hunk 3]
```diff
@@ -7,7 +7,7 @@ end
 
 # user generates detector metadata in code
 function simulate_fields(detector_metadata::AbstractString, environment_settings::PropDict, simulation_settings::PropDict; overwrite::Bool = false)    
-    simulate_fields(propdict(detector_metadata), environment_settings, simulation_settings; overwrite)
+    simulate_fields(readprops(detector_metadata), environment_settings, simulation_settings; overwrite)
```
Comment (oschulz) [inline src/detector.jl]: Let's use `LegendDataManagement.readlprops`, is has some additional features, and we should be able to drop the PropDicts dependency then.

**Review by oschulz — CHANGES_REQUESTED** [review]

File: src/detector.jl (hunk 3 above)
Comment (fhagemann) [reply src/detector.jl]: I don’t think we can drop the PropDicts dependency, as a lot of functions take PropDict as input

Comment (oschulz) [reply src/detector.jl]: Ooops, right!

Comment (oschulz) [reply src/detector.jl]: Still, let's use `readlprops`.

Comment (fhagemann) [reply src/detector.jl]: Done in 4c4d216

**Comment by fhagemann** [comment]
I also had to update the `Project.toml` because `readlprops` was introduced only in `LegendDataManagement@0.2.8`.

**Review by oschulz — APPROVED** [review]

**Pull request merged**